### PR TITLE
Added VersioningDirectives

### PR DESCRIPTION
**Pull request by kshakir**
Also some cleanup of OpenAM directives.

### CODE REVIEW

File: src/test/scala/org/broadinstitute/dsde/vault/common/directives/VersioniingDirectivesSpec.scala [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.{Matchers, FreeSpec}
+import spray.http._
+import spray.http.StatusCodes._
+import spray.httpx.SprayJsonSupport._
+import spray.json.DefaultJsonProtocol._
+import spray.routing.HttpService
+import spray.testkit.ScalatestRouteTest
+
+class VersioniingDirectivesSpec extends FreeSpec with Matchers with ScalatestRouteTest with ScalaFutures with CommonDirectivesSpec with HttpService {
```
Comment (kshakir) [inline src/test/scala/org/broadinstitute/dsde/vault/common/directives/VersioniingDirectivesSpec.scala]: Oops. Need to fix "Version***_ii**_*ng".

File: src/main/scala/org/broadinstitute/dsde/vault/common/openam/OpenAMClient.scala [hunk 2]
```diff
@@ -6,7 +6,7 @@ import spray.client.pipelining._
 import spray.http.Uri
 import spray.httpx.SprayJsonSupport._
 
-import scala.concurrent.{Awaitable, Await}
+import scala.concurrent.{Future, Awaitable, Await}
```
Comment (rushtong) [inline src/main/scala/org/broadinstitute/dsde/vault/common/openam/OpenAMClient.scala]: Some unused imports (Await, Awaitable, and on the line below too.)

**Comment by rushtong** [comment]
Looks good. Squash if you like and then I'll merge it.

**Comment by kshakir** [comment]
:+1: Rebased into two commits. Ready for merging.

**Pull request merged**